### ISSUE TITLE
Create a community "membership directory"

### ISSUE THREAD

**Issue by pkfrank** [forem/forem#11714]
**Is your feature request related to a problem? Please describe.**
As a Forem member, there's no intuitive and elegant way to explore who else is part of the community.

**Describe the solution you'd like**
A "membership directory" where I can browse other registered users of the community.

Stretch features include:
- Ability to filter people by keyword search
- Ability to filter people by other options (location, etc)
    - Basically any "profile field" set by a Forem Admin

**Describe alternatives you've considered**
You can use a search filtered down to users ([example](https://dev.to/search?q=&filters=class_name:User)) but it's not all that great.

**Additional context**
This would be especially useful for smaller communities.  It's a very common feature request from Forem Admins.

**Comment by lisasy** [comment]
This feature makes a lot of sense to me and definitely feels like a gap today.

I think that the product design task would be to identify:
- Where does someone access this directory?
- What is the interaction exploring this directory (does it open up in a new page? a modal? what actions can someone take here? Is it ordered based on people you're following or alphabetically? etc)

**Comment by leewynne** [comment]
This sounds great!

**Comment by michael-tharrington** [comment]
I love this idea and want to resurface it as this request recently came up again for the [Web Monetization Community](https://community.webmonetization.org/). Note that in addition to a user directory, they'd also really like to see an org directory with similar attributes, or perhaps have a way to switch between users and orgs in one single directory.

The problem with https://dev.to/search?q=&filters=class_name:User as a workaround is that the user list is limited to 60 max and the available filters (Most Relevant, Newest, Oldest) don't really apply well to users, and don't effect the user list/order when switched — so this user list always looks the same.

I totally agree that directories are an important feature particularly for smaller communities!
